conjure-java-undertow-runtime/src/test/java/com/palantir/conjure/java/undertow/runtime/ConjureExceptionDisconnectTest.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.undertow.runtime;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.github.stefanbirkner.systemlambda.SystemLambda;
+import io.undertow.Undertow;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.SocketTimeoutException;
+import java.net.URL;
+import java.time.Duration;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class ConjureExceptionDisconnectTest {
+
+    private static Undertow server;
+    private static CountDownLatch latch = new CountDownLatch(1);
+
+    @BeforeAll
+    public static void before() {
+        server = Undertow.builder()
+                .addHttpListener(12345, "localhost")
+                .setHandler(new ConjureExceptionHandler(
+                        _e -> {
+                            Thread.sleep(Duration.ofMillis(50).toMillis());
+                            throw new IOException();
```

Comment:
I tried doing a simple write to the stream so that when it is closed it throws, but then it got mad at me for writing on the IO thread. So then I just figured I would throw, and then it got mad that I never setBlocking this test kinda sucks I need to think about it

Reply:
I think I might also be losing sight as to what I am testing, right now it seems that I mostly testing that `ConjureExceptions` write to INFO which this is an elaborate way of doing so. But making a more real setup with conjure and dialogue means I cannot actually get in to the handler chain because `ConjureExceptions` eats the exception and logs.